### ISSUE TITLE
We want the new local image changes will take effect - BUT HOW?

### ISSUE THREAD

**Issue by igaloly** [pulumi/pulumi#5515]
Hi all :)

We have a pipeline of initialization that consists of:
Loop through each microservice -> For each microservice, if specified, build a local docker image -> Deploy the helm chart.

Let's say we did it once. Everything is fine.
Now, after another change in the microservice, we want to rebuild the docker image again and see the changes.

What happens?
Changes do not appear. They appear only after we rollout the deployment of the specific microservice via `kubectl rollout restart deployments [microservice-name] ...` or via `kubectl delete pod [pod-name]`.

It seems like there is no recognization of changes in the docker image.
Fine by us.

1. If you have some solution to make it work - it'll be great.
2. If you don't have, then we want to delete the pod / reset the deployment programmatically after `pulumi up` is finished (or in some other initialization lifecycle point that'll bring the desired result - image refresh) - And would love to get some help to do that :)

**Comment by leezen** [comment]
Could you please provide an example of how you're doing this? Are you managing the image and build w/in Pulumi itself?

**Comment by igaloly** [comment]
> Could you please provide an example of how you're doing this? Are you managing the image and build w/in Pulumi itself?

Hi, thanks for replying :)

We have a yaml file that's divided into several microservices and their configuration.
We are looping through the yaml, and for each service, If the configuration consists with a key named 'localPath', we get the value from the key - that's the path of the `Dockerfile` inside microservice's dir - and use this path to build local docker image in the `pulumi up` like that:
![image](https://user-images.githubusercontent.com/38460810/95302602-d95a1280-088a-11eb-992d-5a4db68ca22b.png)

After that, we deploy the helm chart using `new k8s.helm.v3.Chart`, which gets the image name in `LocalChartOpts`, and passing it to `template.spec.containers[0].image` as a programmatic value from `Values`, like that:
![image](https://user-images.githubusercontent.com/38460810/95303802-58037f80-088c-11eb-8695-9fac8f694800.png)

That way, we build the docker image in the pulumi pipeline and deploying the helm chart using the built image.

Basically, I would like to get the most fast-convenient way to work on one-or-more microservices locally, when we are initializing the clusters via pulumi.

**Comment by leezen** [comment]
Thanks for the details. If you're flowing the image through, I would expect this to actually cause the chart to update. Is `Values.container.image` coming from the `imageName` output of `dockerImage`? What does your diff show?

**Comment by RobbieMcKinstry** [comment]
Hi @igaloly! Are you still encountering this issue? As Lee wrote, it would help if we could have a little more information.

> Is Values.container.image coming from the imageName output of dockerImage? What does your diff show?

Since we haven't heard back from you, I'll go ahead and close this issue for now. We can reopen if you're still encountering the problem.
